test/runnable/test17924.d

```diff
@@ -0,0 +1,25 @@
+// PERMUTE_ARGS:
+
+void main()
+{
+    int x = 0;
+    int y = 1;
+    assert((x ?: y) == y);
+    assert((y ?: x) == y);
+
+    int* p;
+    int* u;
+    assert((p ?: u) == u);
+    assert((u ?: p) == u);
```

Comment:
What's the point of these two tests ? The two int* are null.

Reply:
Yeah, you don't know if it "returned" `u` or `p` because both are `null`.

Reply:
It was a slip on my part. That is actually an int.

Reply:
I thought you've forgot to assign `u` to something.